src/components/QueryEditor.tsx

```diff
@@ -494,6 +500,38 @@ const QueryEditor: typeof window.roamjs.extension.queryBuilder.QueryEditor = ({
         <style>{`.roamjs-query-custom-enabled .bp3-control.bp3-switch .bp3-control-indicator-child:first-child {
     height: 0;
 }`}</style>
+        <div>
+        {isEditingLabel ? (
+          <InputGroup
+            value={label}
+            onChange={(e) => setLabel(e.target.value)}
+            onKeyDown={(e) => {
+              if (e.key === "Enter") {
+                updateBlock({ uid: parentUid, text: `{{query block:${label}}}` });
+                setIsEditingLabel(false);
+              }
+            }}
+            autoFocus
+            rightElement={
+              <Button
+                minimal
+                icon={"confirm"}
+                onClick={() => {
+                  updateBlock({ uid: parentUid, text: `{{query block:${label}}}` });
```

Comment:
what does this UX look like? does the component flicker or is it smooth?

Reply:
Seems smooth.

https://user-images.githubusercontent.com/3792666/219730603-51b4925c-158c-4eab-96a1-4d339b28d82f.mp4